File: 2016/Prototypes/ShooterTestProgram/src/org/usfirst/frc/team3081/robot/subsystems/ShooterSubsystem.java
package org.usfirst.frc.team3081.robot.subsystems;

import org.usfirst.frc.team3081.robot.RobotMap;

import edu.wpi.first.wpilibj.Jaguar;
import edu.wpi.first.wpilibj.SpeedController;
import edu.wpi.first.wpilibj.command.Subsystem;

/**
 *
 */
public class ShooterSubsystem extends Subsystem {
    
    // Put methods for controlling this subsystem
    // here. Call these from Commands.

	private final SpeedController left = new Jaguar(RobotMap.RIGHT_SHOOTER);
	private final SpeedController right = new Jaguar(RobotMap.LEFT_SHOOTER);
	
    public void initDefaultCommand() {
        // Set the default command for a subsystem here.
        //setDefaultCommand(new MySpecialCommand());
    }
    
    public void stop() {
    	this.set(0.0, 0.0);
    }
    
    public void set(double leftspeed, double rightspeed) {
    	left.set(leftspeed);
    	right.set(rightspeed);
    }
}

File: 2017/2017Robot/src/Subsystems/ShooterSubsystem.cpp
#include "ShooterSubsystem.h"
#include "../RobotMap.h"
#include <Jaguar.h>

constexpr double SHOOT_POWER = 1.0;

ShooterSubsystem::ShooterSubsystem() : Subsystem("ShooterSubsystem") {
	this->jag = new Jaguar(SHOOTER_CONTROLLER);
}

void ShooterSubsystem::InitDefaultCommand() {
	// Set the default command for a subsystem here.
	// SetDefaultCommand(new MySpecialCommand());
}

// Put methods for controlling this subsystem
// here. Call these from Commands.
void ShooterSubsystem::Shoot() {
	this->jag->Set(SHOOT_POWER);
}

void ShooterSubsystem::Stop() {
	this->jag->Set(0);
}
